targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F469xI/TARGET_DISCO_F469NI/PeripheralPins.c

```diff
@@ -343,3 +343,24 @@ const PinMap PinMap_CAN_TD[] = {
     {PB_13, CAN_2, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF9_CAN2)},
     {NC,    NC,    0}
 };
+
+
+const PinMap PinMap_QSPI_DATA[] = {
+    {PF_6,  0, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF9_QSPI)},
+    {PF_7,  0, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF9_QSPI)},
+    {PF_8, 0, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF10_QSPI)},
+    {PF_9,  0, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF10_QSPI)},
+
```

Comment:
Hello, thank you for this.
The question is: do we split the 4 data lines into 4 separated buffers or not ?
If not, could we use STM_PIN_DATA_EXT to add the Dx value (D0 / D1 / D2 /D3) ?

The bank number information is also required (I suggest to use QSPIBK_1 and QSPIBK_2 the second field) for Data and NSS buffers (not needed for CLK buffer).

Cheers

Reply:
Or use the 2nd field for the peripheral name (QUADSPI) and use other fields for the bank and data line numbers:

```
{PF_6,  QUADSPI, STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF9_QSPI, 1, 3, 0)}, // BK1 / D3
{PF_7,  QUADSPI, STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF9_QSPI, 1, 3, 0)}, // BK1 / D2
```
...
Because on some devices there are 2 banks to control 2 QSPI flash.

Reply:
I am missing some context.  can you reference some example or driver that does this - setting the bank + Dx value ?

> Or use the 2nd field for the peripheral name (QUADSPI) and use other fields for the bank and data line numbers:

why peripheral need would be part of the pin configuration?

Reply:
It might be easier, if I provide you write access to quickly do the requested changes to my branch so would appear here ?

Reply:
In fact the second field is always used for the peripheral name: ADC_1, PWM_2, SPI_1 etc... My remark was just to do the same for the QSPI.

Example with PWM:
{PE_10, PWM_1, STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 2, 1)},  // TIM1_CH2N
In the STM_PIN_DATA_EXT we use the fourth field to indicate the channel (here 2) and the fifth field used to indicate if the channel is reverted (here value = 1) or not (value = 0).

I was thinking doing the same for the QSPI pins: fourth field used to indicate the bank number and the fifth field to indicate the data line. But thinking again, I am not sure if the data line information is really required? Maybe not after all.

Reply:
Not certain what data line hold as information.

Can you send PR to my fork to get this bank info in place, or point me to the directions how this dual bank works and should be defined for pins? Or share the code snippet here?